pkg/golinters/gomnd.go

```diff
@@ -9,10 +9,23 @@ import (
 )
 
 func NewGoMND(settings *config.GoMndSettings) *goanalysis.Linter {
+	// The constant is only used to force the analyzer name to use the same name as the linter.
+	// This is required to avoid displaying the analyzer name inside the issue text.
+	//
+	// Alternative names cannot help here because of the linter configuration that uses `gomnd` as a name.
+	// The complexity of handling alternative names at a lower level (i.e. `goanalysis.Linter`) isn't worth the cost.
+	// The only way to handle it properly is to deprecate and "duplicate" the linter and its configuration,
```

Comment:
I think it would be enough to duplicate the configuration and pick the one that's non empty or the newest one. Then we can use the same deprecation cycle as with linters to show warnings if you use the `gomnd` key in the config.

I know they're public but I suspect we don't see `NewGoMND` as a public API (given that `NewErr113` was just changed so we could just accept multiple configuration:

```go
func NewGoMND(settings *config.MndSettings, deprecatedSettings *config.GoMndSettings) *goanalysis.Linter
```

Reply:
In all cases, the public API of golangci-lint can be broken, because golangci-lint should not be used as a library.

I put quotes around `duplicate`, because, yes I was not suggesting to duplicate the code but to "register" a new linter with the same code.

Reply:
In the line after, I said "for now, I don't know if it's worth the cost."

I feel like you thing it is worth the cost, so I will "duplicate" the linter.

Reply:
I will split the PR into 2 PRs because the deprecation of gomnd is not the same topic as the simple rename of an analyzer. I don't feel comfortable having 2 differents topics inside the one PR.

Reply:
https://github.com/golangci/golangci-lint/pull/4572